File: go/tree/979. 在二叉树中分配硬币.go
package tree

func distributeCoins(root *TreeNode) int {
	var dfs func(*TreeNode) int
	var res int
	dfs = func(root *TreeNode) int {
		if root == nil {
			return 0
		}
		left := dfs(root.Left)
		right := dfs(root.Right)

		if left < 0 {
			res += (-1 * left)
		} else {
			res += left
		}
		if right < 0 {
			res += (-1 * right)
		} else {
			res += right
		}
		return root.Val + left + right - 1
	}
	dfs(root)
	return res
}

File: python/树/979. 在二叉树中分配硬币.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def distributeCoins(self, root: TreeNode) -> int:
        import math
        self.res = 0
        def dfs(root):
            if not root:return 0
            left = dfs(root.left)
            right = dfs(root.right)
            self.res += abs(left) + abs(right)
            return root.val + left + right -1
        dfs(root)
        return self.res
